### PR TITLE
Fixes #32113 - Show errors on form submission

### PR DESCRIPTION
**Pull request by xprazak2**
Needs https://github.com/theforeman/foreman/pull/8318

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
[test foreman_templates]

File: webpack/components/NewTemplateSync/components/NewTemplateSyncForm/NewTemplateSyncForm.js [hunk 1]
```diff
@@ -74,9 +74,9 @@ const NewTemplateSyncForm = ({
           )(values[syncType]),
           message: `Templates were ${syncType}ed.`,
           item: 'TemplateSync',
-        }).then(args => {
-          history.replace({ pathname: '/template_syncs/result' });
-        });
+          actions,
+          successCallback: () => history.replace({ pathname: '/template_syncs/result' })
+        })
```
Comment (amirfefer) [inline webpack/components/NewTemplateSync/components/NewTemplateSyncForm/NewTemplateSyncForm.js:79]: nit: can you please extract the onSubmit callback outside ?
makes it more readable, plus it won't be re-render every rendering.
```js
const onFormSubmit = (values, actions) => {
    const url = syncType === 'import' ? importUrl : exportUrl;
    return submitForm({
      url,
      values: compose(
         addLocParams,
        addOrgParams
      )(values[syncType]),
      message: `Templates were ${syncType}ed.`,
      item: 'TemplateSync',
      actions,
      successCallback: () => history.replace({ pathname: '/template_syncs/result' })
    })
  return (
    <ForemanForm onSubmit={onFormSubmit} /> 
   )
```

**Comment by xprazak2** [comment]
Updated.

**Review by ezr-ondrej — APPROVED** [review]
Works great, thanks @xprazak2 !

**Comment by ezr-ondrej** [comment]
Cant merge though.

**Comment by tbrisker** [comment]
Thanks @xprazak2 , merging based on reviews.

**Pull request merged**